Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs

```diff
@@ -271,7 +258,22 @@ internal BrowserCommandResult<LoginResult> PassThroughLogin(Uri uri)
             {
                 driver.Navigate().GoToUrl(uri);
 
-                WaitForLoginPage();
+                WaitForMainPage(60.Seconds(),
+                    _ =>
+                    {
+                        //determine if we landed on the Unified Client Main page
+                        var isUCI = driver.HasElement(By.XPath(Elements.Xpath[Reference.Login.CrmUCIMainPage]));
+                        if (isUCI)
+                        {
+                            driver.WaitForPageToLoad();
+                            driver.WaitForTransaction();
+                        }
+                        else
+                            //else we landed on the Web Client main page or app picker page
+                            SwitchToDefaultContent(driver);
+                    },
+                    () => driver.Throw("Load Main Page Fail.")
+                );

```

Comment:
WaitForLoginPage is just a variation of WaitForMainPage with more time & throw an Error if this is not shown.